policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,227 @@
+use crate::{
+    http_route::{ParentReference, RouteBinding},
+    resource_id::ResourceId,
+};
+use ahash::{AHashMap as HashMap, AHashSet as HashSet};
+use chrono::offset::Utc;
+use linkerd_policy_controller_k8s_api::{self as k8s, gateway, ResourceExt};
+use parking_lot::RwLock;
+use std::{collections::hash_map::Entry, sync::Arc};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+const POLICY_API_GROUP: &str = "policy.linkerd.io";
+const POLICY_API_VERSION: &str = "policy.linkerd.io/v1alpha1";
+const STATUS_CONTROLLER_NAME: &str = "policy.linkerd.io/status-controller";
+
+pub type SharedIndex = Arc<RwLock<Index>>;
+
+pub struct Controller {
+    client: k8s::Client,
+    updates: UnboundedReceiver<Update>,
+}
+
+pub struct Index {
+    updates: UnboundedSender<Update>,
+
+    http_routes: HashMap<ResourceId, RouteBinding>,
+    servers: HashSet<ResourceId>,
+}
+
+pub struct Update {
+    id: ResourceId,
+    patch: k8s::Patch<serde_json::Value>,
+}
+
+impl Controller {
+    pub fn new(client: k8s::Client, updates: UnboundedReceiver<Update>) -> Self {
+        Self { client, updates }
+    }
+
+    pub async fn process_updates(mut self) {
+        let patch_params = k8s::PatchParams::apply("policy.linkerd.io");
+
+        // todo: If an update fails we should figure out a requeueing strategy
+        while let Some(Update { id, patch }) = self.updates.recv().await {
+            let api =
+                k8s::Api::<k8s::policy::HttpRoute>::namespaced(self.client.clone(), &id.namespace);
+
+            // todo: Do we need to consider a timeout here?
```

Comment:
do we want to address either of these TODOs before merging this PR, or do we plan on addressing them in the future?

Reply:
We'll address them in the future. We can be more relaxed with failures for now since nothing is consuming the `status` field on HTTPRoutes. However, once the policy controller does start consuming this field, we should consider the different failure scenarios so that we know `status` is as correct as possible.